### PR TITLE
Misc fixes7 - MOC and feedparser improvements (hopefully)

### PR DESCRIPTION
**Pull request by mitchdowney**

### CODE REVIEW

**Comment by mitchdowney** [comment]
These PRs are ready for review I think

File: Podverse/PodcastsTableViewController.swift [hunk 1]
```diff
@@ -229,13 +237,17 @@ extension PodcastsTableViewController:UITableViewDelegate, UITableViewDataSource
...
-        let podcastToEditOid = self.subscribedPodcastsArray[indexPath.row].objectID
+        
         let podcastToEditFeedUrl = self.subscribedPodcastsArray[indexPath.row].feedUrl
         
         let deleteAction = UITableViewRowAction(style: .default, title: "Delete", handler: {action, indexpath in
             self.subscribedPodcastsArray.remove(at: indexPath.row)
             tableView.deleteRows(at: [indexPath], with: .fade)
-            PVDeleter.deletePodcast(feedUrl: podcastToEditFeedUrl, moc: self.moc)
+            
+            DispatchQueue.global().async {
```
Comment (kreonjr) [inline Podverse/PodcastsTableViewController.swift]: Why not dispatching to global and creating a privateMOC inside the deletePodcast method? Tis way you can avoid dispatching every time you call delete

File: Podverse/PodcastsTableViewController.swift [hunk 2]
```diff
@@ -285,28 +297,32 @@ extension PodcastsTableViewController {
...
     
     func refreshParsingStatus(_ notification:Notification) {
-        let total = self.parsingPodcastsList.urls.count
-        let currentItem = self.parsingPodcastsList.currentlyParsingItem
+        let total = self.parsingPodcasts.urls.count
+        let currentItem = self.parsingPodcasts.currentlyParsingItem
         
-        DispatchQueue.main.async {
-            if total > 0 && currentItem < total {
+        if total > 0 && currentItem < total {
```
Comment (kreonjr) [inline Podverse/PodcastsTableViewController.swift]: Just curious, why is this different that dispatching once and having the if statements inside the dispatch?

File: Podverse/PodcastsTableViewController.swift (hunk 1 above)
Comment (mitchdowney) [reply Podverse/PodcastsTableViewController.swift]: Good catch, I'm updating all the delete methods to dispatch to global inside them.

File: Podverse/PodcastsTableViewController.swift (hunk 2 above)
Comment (mitchdowney) [reply Podverse/PodcastsTableViewController.swift]: I had this notion that, if you dispatch to the main queue, you should only dispatch the parts that will directly change the UI, whereas evaluating that conditional logic wouldn't directly change the UI, so I thought that should be kept off the main thread.

It seems like I'm splitting hairs here though. I'll go ahead and move the conditional into the main thread.

**Comment by mitchdowney** [comment]
Ready for re-review

**Pull request merged**